src/core_atmosphere/physics/mpas_atmphys_driver_radiation_sw.F

```diff
@@ -949,13 +950,34 @@ subroutine driver_radiation_sw(itimestep,configs,mesh,state,time_lev,diag_physic
 
  call mpas_pool_get_config(configs,'config_o3climatology' ,config_o3climatology)
  call mpas_pool_get_config(configs,'config_radt_sw_scheme',radt_sw_scheme      )
+ call mpas_pool_get_config(configs,'config_dt'            ,dt                  )
 
  xtime_m = xtime_s/60.
 
+ if (itimestep == 1 .and. dt_radtsw == dt .and. &
+     (mpas_cpl%role_is(ROLE_INTEGRATE) .or. mpas_cpl%role_is(ROLE_RADIATION))) then
+     ! When running lagged radiation with dt == radt, the 2nd and 3rd timesteps would receive
+     ! SW radiation computed at 1 radt behind.
+     call mpas_log_write('')
+     call mpas_log_write('WARNING: Lagged radiation with dt==radt is not fully supported.')
+     call mpas_log_write('')
+ end if
+
  if (mpas_cpl%role_includes(ROLE_RADIATION)) then
 ! This should be OMP MASTER with barrier afterwards or OMP SINGLE, since declin and solcon
 ! are global variables in mpas_atmphys_vars.F and race conditions may occur otherwise!
 !$OMP SINGLE
+
+!... convert the radiation time_step to minutes:
+ radt = dt_radtsw/60.
+
+!... Adjust effective solar time for lagged radiation
+ if (itimestep == 2) then
+    xtime_m = xtime_m - dt/60. + radt ! Special radiation step after init
+ elseif (itimestep > 2) then
+    xtime_m = xtime_m + radt
+ end if
```

Comment:
Should this code only be executed for lagged-radiation runs?

Reply:
Yes, good catch. I'll correct the code.

As far as LW goes, RRTMG_LW does not seem to be dependent on time.